src/node/mini_miner.cpp

```diff
@@ -0,0 +1,365 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <node/mini_miner.h>
+
+#include <consensus/amount.h>
+#include <policy/feerate.h>
+#include <primitives/transaction.h>
+#include <timedata.h>
+#include <util/check.h>
+#include <util/moneystr.h>
+
+#include <algorithm>
+#include <numeric>
+#include <utility>
+
+namespace node {
+
+MiniMiner::MiniMiner(const CTxMemPool& mempool, const std::vector<COutPoint>& outpoints)
+{
+    LOCK(mempool.cs);
+    // Find which outpoints to calculate bump fees for.
+    // Anything that's spent by the mempool is to-be-replaced
+    // Anything otherwise unavailable just has a bump fee of 0
+    for (const auto& outpoint : outpoints) {
+        if (!mempool.exists(GenTxid::Txid(outpoint.hash))) {
+            // This UTXO is either confirmed or not yet submitted to mempool.
+            // If it's confirmed, no bump fee is required.
+            // If it's not yet submitted, we have no information, so return 0.
+            m_bump_fees.emplace(outpoint, 0);
+            continue;
+        }
+
+        // UXTO is created by transaction in mempool, add to map.
+        // Note: This will either create a missing entry or add the outpoint to an existing entry
+        m_requested_outpoints_by_txid[outpoint.hash].push_back(outpoint);
+
+        if (const auto ptx{mempool.GetConflictTx(outpoint)}) {
+            // This outpoint is already being spent by another transaction in the mempool. We
+            // assume that the caller wants to replace this transaction and its descendants. It
+            // would be unusual for the transaction to have descendants as the wallet won’t normally
+            // attempt to replace transactions with descendants. If the outpoint is from a mempool
+            // transaction, we still need to calculate its ancestors bump fees (added to
+            // m_requested_outpoints_by_txid below), but after removing the to-be-replaced entries.
+            //
+            // Note that the descendants of a transaction include the transaction itself. Also note,
+            // that this is only calculating bump fees. RBF fee rules should be handled separately.
+            CTxMemPool::setEntries descendants;
+            mempool.CalculateDescendants(mempool.GetIter(ptx->GetHash()).value(), descendants);
+            for (const auto& desc_txiter : descendants) {
+                m_to_be_replaced.insert(desc_txiter->GetTx().GetHash());
+            }
+        }
+    }
+
+    // No unconfirmed UTXOs, so nothing mempool-related needs to be calculated.
+    if (m_requested_outpoints_by_txid.empty()) return;
+
+    // Calculate the cluster and construct the entry map.
+    std::vector<uint256> txids_needed;
```

Comment:
https://cirrus-ci.com/task/4526522288046080
seems like the tidy job wants you to add this:
```suggestion
    std::vector<uint256> txids_needed;
    txids_needed.reserve(m_requested_outpoints_by_txid.size());
```